testsuite/utils.py

```diff
@@ -136,3 +137,11 @@ def fire_requests(client, max_requests, period, grace_requests=0, iterations=1,
         assert httpx.get(url).status_code == 429, f"Iteration {iteration + 1} failed to start limiting"
         sleep(period)
         assert httpx.get(url).status_code == 200, f"Iteration {iteration + 1} failed to reset limits"
+
+
+def extract_from_response(response, *args):
+    """Extracts the UUID expectation from the request metadata"""
+    result = json.loads(response.json()["headers"]["Auth-Json"])["auth"]
+    for arg in args:
+        result = result.get(arg)
```

Comment:
I think `assert` or plain `result[arg]` here will produce more comprehensive error message, if any key does not exist

Reply:
agreed, I will use `result[arg]`